### PR TITLE
38/link command should not execute npm install if it was done before

### PR DESCRIPTION
**Pull request by HaidarJbeily7**
## Changes
- Added a check if the npm install was already executed before. I implemented it by checking where the package-lock.json file exists.

---

## PR-Codex overview
This PR focuses on enhancing the `link` command by preventing unnecessary reinstallation of npm packages if the `package-lock.json` file already exists. It also includes a new test case to verify this behavior.

### Detailed summary
- Added a check for the existence of `package-lock.json` before running `npm install`.
- Introduced a new test to confirm that `package-lock.json` is not modified on subsequent runs when it exists.
- Adjusted code structure to improve readability and maintainability.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by maxonfjvipon** [comment]
@HaidarJbeily7 thanks for your contribution, but it would be nice to add a test that confirms that your changes work and don't break anything

File: eo2js/test/commands/link.test.js [hunk 1]
```diff
@@ -39,4 +39,20 @@ describe('link', function() {
     assertFilesExist(link('--tests'), project, ['node_modules/mocha'])
     done()
   })
+  it('should not reinstall npm packages if package-lock.json exists', function(done) {
+    // First run to create initial files
+    link()
+    
+    // Get modification time of package-lock.json
```
Comment (maxonfjvipon) [inline eo2js/test/commands/link.test.js]: @HaidarJbeily7 inline comments and empty lines are not necessary

**Review by maxonfjvipon — CHANGES_REQUESTED** [review]

File: eo2js/test/commands/link.test.js (hunk 1 above)
Comment (HaidarJbeily7) [reply eo2js/test/commands/link.test.js]: Removed! @maxonfjvipon

File: eo2js/src/commands/link.js [hunk 2]
```diff
@@ -56,18 +56,21 @@ const link = function(options) {
     path.resolve(project, 'package.json'),
     JSON.stringify(pckg(options))
   )
-  execSync('npm install', {cwd: project})
+  const packageLockPath = path.resolve(project, 'package-lock.json')
```
Comment (maxonfjvipon) [inline eo2js/src/commands/link.js]: @HaidarJbeily7 let's rename this variable to some single word one: https://www.yegor256.com/2015/01/12/compound-name-is-code-smell.html

Maybe something like `lock`?

After this we're good to merge

**Review by maxonfjvipon — CHANGES_REQUESTED** [review]
@HaidarJbeily7 looks good to me, just one simple comment from my side

File: eo2js/src/commands/link.js (hunk 2 above)
Comment (HaidarJbeily7) [reply eo2js/src/commands/link.js]: yup done! Thanks for this article.

**Review by maxonfjvipon — APPROVED** [review]
@HaidarJbeily7 LGTM!

**Comment by maxonfjvipon** [comment]
@rultor merge

**Comment by maxonfjvipon** [comment]
@HaidarJbeily7 thanks!

**Pull request merged**